configure

```diff
@@ -1,9 +1,110 @@
-#!/usr/bin/env ruby
+#!/usr/bin/ruby
+require 'fileutils'
+require 'date'
 
-$current_dir = File.dirname(File.expand_path(__FILE__))
-Dir[File.join($current_dir, "setup/*.rb")].each do |file|
-  require_relative(file)
+def prompt_for_variable(variable, default)
+  return ARGV[0] if variable == '__PROJECT_NAME__' && ARGV.count == 1
+  puts "Please enter a value for #{variable} (default: \"#{default}\")"
+  answer = STDIN.gets.chomp
+  answer.length > 0 ? answer : default
 end
 
-pod_name = ARGV.shift
-Pod::TemplateConfigurator.new(pod_name).run
+def replace_variables_in_files(substitutions)
+  Dir.glob("**/*", File::FNM_DOTMATCH) do |file_name|
+    next if File.directory?(file_name)
+
+    text = ''
+    File.open(file_name, 'rb') do |file|
+      text = file.read
+      substitutions.each do |old, new|
+        text.gsub!(old, new)
+      end
+    end
+    File.open(file_name, 'wb') do |file|
+      file.write(text)
+    end
+  end
+end
+
+def replace_variables_in_file_names(substitutions)
+  Dir.foreach('.') do |file_name|
+    next if file_name == '.' or file_name == '..'
+    new_file_name = file_name.clone
+    substitutions.each do |old, new|
+      new_file_name.gsub!(old, new)
+    end
+    if new_file_name != file_name
+      FileUtils.mv file_name, new_file_name
+    end
+    if File.directory?(new_file_name)
+      Dir.chdir(new_file_name) do
+        replace_variables_in_file_names(substitutions)
+      end
+    end
+  end
+end
+
+module Env
+  class << self
+    ENV_VARIABLE_PREFIX = 'SMT'
+
+    def fetch_smt(template_var_name, *args, &b)
+      environment_var_name = name_for(template_var_name.to_s)
+      ENV.fetch(environment_var_name) do
+        fallback(environment_var_name, template_var_name, *args, &b)
+      end
+    end
+
+    def date_format_string
+      @date_format_string ||=
+        fetch_smt(:DATE_FORMAT_STRING, '%b %-d, %Y')
+    end
+
+    def name_for(template_var_name)
+      "#{ ENV_VARIABLE_PREFIX }_#{ sanitize(template_var_name) }"
+    end
+
+    private
+
+    def fallback(environment_var_name, template_var_name, *args)
+      if block_given?
+        yield(template_var_name, *args)
+      else
+        ENV.fetch(environment_var_name, *args)
+      end
+    end
+
+    def sanitize(template_var_name)
+      template_var_name.upcase.gsub(/\W/, '_').gsub(/^_+|_+$/, '')
+    end
+  end
+end
+
+##
+## MAIN
+##
+
+substitutions = {
+  '__PROJECT_NAME__' => 'MyProject',
+  '__ORGANIZATION NAME__' => 'Awesome Org',
+  'com.AN.ORGANIZATION.IDENTIFIER' => 'com.awesome',
+  '__AUTHOR NAME__' => 'Mr McAwesome',
+  '__TODAYS_DATE__' => Date.today.strftime(Env.date_format_string),
+  '__TODAYS_YEAR__' => "#{Date.today.year}",
+  '__GITHUB_USERNAME__' => 'awesome_octocat'
```

Comment:
Do people really care about adding the org name and bundle ID? Also, it should just skip the date + year.

This CLI output feels really mechanical:
<img width="815" alt="screen shot 2017-06-25 at 07 38 22" src="https://user-images.githubusercontent.com/49038/27515800-37867556-597a-11e7-8a79-93d7d1eb8e27.png">

In contrast to 

<img width="689" alt="screen shot 2017-06-25 at 07 45 49" src="https://user-images.githubusercontent.com/49038/27515814-71b863ba-597a-11e7-9e57-b2b2531300bf.png">

<img width="1280" alt="screen shot 2017-06-25 at 07 47 15" src="https://user-images.githubusercontent.com/49038/27515826-df9fa794-597a-11e7-82bb-c37a0c195d30.png">


I think we can improve the onboarding  a bit here. Though, that said, I'm probably going to delete the guide unless you want to update this: https://guides.cocoapods.org/making/using-pod-lib-create.html